roles/upgrade/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+- name: "Stop the OpenLDAP Server"
+  ansible.builtin.systemd:
+    name: slapd
+    daemon_reload: true
+    enabled: true
+    state: stopped
+
+- name: Backup LDAP DB
+  ansible.builtin.command: slapcat -l /tmp/ldap-backup.ldif
+  become: yes
+  args:
+    creates: /tmp/ldap-backup.ldif
+
+- name: Clear OpenLDAP DB directory
+  ansible.builtin.file:
+    path: /var/lib/ldap
+    state: absent
+  become: yes
+
+- name: Recreate OpenLDAP DB directory
+  ansible.builtin.file:
+    path: /var/lib/ldap
+    state: directory
+    owner: openldap
+    group: openldap
+    mode: "0750"
+  become: yes
+
+- name: Install specific version of OpenLDAP (slapd)
```

Comment:
before installing new version make sure to disable or mask the auto-start
```yaml
- name: Mask slapd to prevent auto-start during upgrade
  ansible.builtin.systemd:
    name: slapd
    masked: true
```

and ummask after install and before starting the ldap
```yaml
- name: Unmask slapd after upgrade
  ansible.builtin.systemd:
    name: slapd
    masked: false
```